flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/sink/KafkaRecordSerializationSchemaBuilderTest.java

```diff
@@ -256,6 +267,102 @@ public void testSerializeRecordWithTimestamp() {
         assertThat(recordWithInvalidTimestamp.timestamp()).isNull();
     }
 
+    @Test
+    public void testGetLineageDatasetFacetsWhenTopicSelectorNotKafkaTopicsIdentifierProvider() {
+        SerializationSchema<String> serializationSchema = new SimpleStringSchema();
+        KafkaRecordSerializationSchema<String> schema =
+                KafkaRecordSerializationSchema.builder()
+                        .setTopicSelector(mock(TopicSelector.class))
+                        .setValueSerializationSchema(serializationSchema)
+                        .setKeySerializationSchema(serializationSchema)
+                        .build();
+
+        assertThat(((KafkaDatasetFacetProvider) schema).getKafkaDatasetFacet()).isEmpty();
+    }
+
+    @Test
+    public void testGetLineageDatasetFacetsWhenNoTopicsIdentifiersFound() {
+        TopicSelector topicSelector =
+                mock(
+                        TopicSelector.class,
+                        withSettings().extraInterfaces(KafkaDatasetIdentifierProvider.class));
+        when(((KafkaDatasetIdentifierProvider) topicSelector).getDatasetIdentifier())
+                .thenReturn(Optional.empty());
```

Comment:
I haven't looked too closely at the tests. But a high-level comment: In Flink, we don't use mockito (anymore). The main idea is that we use interfaces (as you did) and then just explicitly create our MockImplementation.

```
class MockTopicSelector implements TopicSelector, KafkaDatasetIdentifierProvider {
  KafkaDatasetIdentifier id; // init with ctor or factory method

  KafkaDatasetIdentifier getDatasetIdentifier() { return id; }
}
```